File: backend/LendACarAPI/DTOs/UserDto.cs
using LendACarAPI.Data.Models;
using System.ComponentModel.DataAnnotations;

namespace LendACarAPI.DTOs
{
    public class UserDto
    {
        [Required]
        public string? FirstName { get; set; }

        [Required]
        public string? LastName { get; set; }

        [Required]
        [Phone]
        public string? PhoneNumber { get; set; }

        [Required]
        public DateTime BirthDate { get; set; }

        public int CityId { get; set; }
        public City? City { get; set; }

        [Required]
        [EmailAddress]
        public string? EmailAddress { get; set; }

        [Required]
        public string? Username { get; set; }
        public double AverageRating { get; set; } = 0.0;
    }
}

File: frontend/src/app/Models/UserDto.ts
import {City} from './City';

export interface UserDto {
  id:number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  birthDate: string;
  cityId: number;
  city: City;
  emailAddress: string;
  username: string;
  averageRating: number;
}
